logql/v2/parser_test.go

```diff
@@ -1477,3 +1528,13 @@ func trimOutput(s string) string {
 	}
 	return s
 }
+
+func TestQuotesEncode(t *testing.T) {
+	expr, err := ParseExpr(`{app="test"}`)
+	require.NoError(t, err)
+	require.Equal(t, `{app="test"}`, expr.String())
+
+	expr, err = ParseExpr("{app=\"test\"}|~`key\":\"val\"`")
+	require.NoError(t, err)
+	require.Equal(t, `{app="test"} |~ "key\":\"val\""`, expr.String())
+}
```

Comment:
Can we eliminate testify usage here? This repo is under shared maintenance with many teams and we don't have any vote on changing the way we write tests.

Reply:
btw: I tend to write flat tests rather than test cases arrays (aka table-driven tests), that's a personal preference, but since the norm here seems to be test cases arrays, I can definitely comply with it.